src/firebase.js

```diff
@@ -3,330 +3,13 @@
 
   angular.module("firebase")
 
-    // The factory returns an object containing the value of the data at
-    // the Firebase location provided, as well as several methods. It
-    // takes one or two arguments:
-    //
-    //   * `ref`: A Firebase reference. Queries or limits may be applied.
-    //   * `config`: An object containing any of the advanced config options explained in API docs
-    .factory("$firebase", [ "$log", "$firebaseUtils", "$firebaseConfig",
-      function ($log, $firebaseUtils, $firebaseConfig) {
-        function AngularFire(ref, config) {
-          // make the new keyword optional
-          if (!(this instanceof AngularFire)) {
-            return new AngularFire(ref, config);
-          }
-          this._config = $firebaseConfig(config);
-          this._ref = ref;
-          this._arraySync = null;
-          this._objectSync = null;
-          this._assertValidConfig(ref, this._config);
-        }
+    /** @deprecated */
+    .factory("$firebase", function() {
+      return function() {
+        throw new Error('$firebase has been removed. You may instantiate $FirebaseArray and $FirebaseObject ' +
+        'directly now. For simple write operations, just use the Firebase ref directly. ' +
+        'See CHANGELOG for details and migration instructions: https://www.firebase.com/docs/web/changelog.html');
+      };
+    });
```

Comment:
Decided to go ahead and keep an error around as it should reduce upgrade support a bit and hopefully point upgrading users to the correct doc. This assumes we'll link to migration instructions from the changelog.